debezium-core/src/main/java/io/debezium/schema/SchemaFactory.java

```diff
@@ -0,0 +1,293 @@
+import io.debezium.data.EnumSet;
+import io.debezium.data.Json;
+import io.debezium.data.Uuid;
+import io.debezium.data.VariableScaleDecimal;
+import io.debezium.data.Xml;
+import io.debezium.heartbeat.HeartbeatImpl;
+import io.debezium.pipeline.txmetadata.TransactionMonitor;
+import io.debezium.relational.history.ConnectTableChangeSerializer;
+import io.debezium.relational.history.HistoryRecord;
+import io.debezium.util.SchemaNameAdjuster;
+
+/**
+ * A factory for creating {@link SchemaBuilder} structs.
+ *
+ * @author Anisha Mohanty
+ */
+
+public class SchemaFactory {
+
+    /*
+     * Heartbeat schemas
+     */
+    private static final String HEARTBEAT_KEY_SCHEMA_NAME = "io.debezium.connector.common.ServerNameKey";
+    private static final int HEARTBEAT_KEY_SCHEMA_VERSION = 1;
+
+    private static final String HEARTBEAT_VALUE_SCHEMA_NAME = "io.debezium.connector.common.Heartbeat";
+    private static final int HEARTBEAT_VALUE_SCHEMA_VERSION = 1;
+
+    /*
+     * Transaction-related schemas
+     */
+    private static final String TRANSACTION_METADATA_KEY_SCHEMA_NAME = "io.debezium.connector.common.TransactionMetadataKey";
+    private static final int TRANSACTION_METADATA_KEY_SCHEMA_VERSION = 1;
+
+    private static final String TRANSACTION_METADATA_VALUE_SCHEMA_NAME = "io.debezium.connector.common.TransactionMetadataValue";
+    private static final int TRANSACTION_METADATA_VALUE_SCHEMA_VERSION = 1;
+
+    private static final String TRANSACTION_BLOCK_SCHEMA_NAME = "event.block";
+    private static final int TRANSACTION_BLOCK_SCHEMA_VERSION = 1;
+
+    private static final String TRANSACTION_EVENT_COUNT_COLLECTION_SCHEMA_NAME = "event.collection";
+    private static final int TRANSACTION_EVENT_COUNT_COLLECTION_SCHEMA_VERSION = 1;
+
+    /*
+     * Schema history schemas
+     */
+    private static final String SCHEMA_HISTORY_CONNECTOR_SCHEMA_NAME_PREFIX = "io.debezium.connector.";
+
+    private static final String SCHEMA_HISTORY_CONNECTOR_KEY_SCHEMA_NAME_SUFFIX = ".SchemaChangeKey";
+    private static final int SCHEMA_HISTORY_CONNECTOR_KEY_SCHEMA_VERSION = 1;
+
+    private static final String SCHEMA_HISTORY_CONNECTOR_VALUE_SCHEMA_NAME_SUFFIX = ".SchemaChangeValue";
+    private static final int SCHEMA_HISTORY_CONNECTOR_VALUE_SCHEMA_VERSION = 1;
+
+    private static final String SCHEMA_HISTORY_TABLE_SCHEMA_NAME = "io.debezium.connector.schema.Table";
+    private static final int SCHEMA_HISTORY_TABLE_SCHEMA_VERSION = 1;
+
+    private static final String SCHEMA_HISTORY_COLUMN_SCHEMA_NAME = "io.debezium.connector.schema.Column";
+    private static final int SCHEMA_HISTORY_COLUMN_SCHEMA_VERSION = 1;
+
+    private static final String SCHEMA_HISTORY_CHANGE_SCHEMA_NAME = "io.debezium.connector.schema.Change";
+    private static final int SCHEMA_HISTORY_CHANGE_SCHEMA_VERSION = 1;
+
+    /*
+     * Datatype schemas
+     */
+
+    /*
+     * Bits schema
+     */
+    private static final int BITS_SCHEMA_VERSION = 1;
```

Comment:
@ani-sha I'd probably place the `VERSION` side by side with the schema name in the class. Also if we keep the schema, fields, version etc. in the separate class then it would make sense to keep the simple namse so just `SCHEMA_NAME`, `VERSION` etc. without adding the type name as there is no name conflict.

Reply:
Yes, makes sense. Will add the change.